example-code/snippets/src/backend/WorkspaceExamples.test.ts

```diff
@@ -0,0 +1,303 @@
+      interface HardinessRange {
+        minimum: number;
+        maximum: number;
+      }
+
+      const range: HardinessRange = { minimum: 8, maximum: 10 };
+      await iModel.acquireSchemaLock();
+      iModel.saveSettingDictionary("landscapePro/iModelSettings", {
+        "landscapePro/hardinessRange": range,
+      });
+      // __PUBLISH_SECTION_END__
+      const iModelName = iModel.pathName;
+      iModel.close();
+      iModel = StandaloneDb.openFile(iModelName, OpenMode.ReadWrite);
+      
+      // __PUBLISH_SECTION_START__ WorkspaceExamples.QuerySettingDictionary
+      const hardinessRange = iModel.workspace.settings.getObject<HardinessRange>("landscapePro/hardinessRange");
+      // returns { minimum: 8, maximum: 10 }
+      defaultTool = iModel.workspace.settings.getString("landscapePro/ui/defaultTool");
+      // returns "place-koi-pond" as specified by IModelHost.appWorkspace.settings.
+      // __PUBLISH_SECTION_END__
+      expect(hardinessRange).to.deep.equal(range);
+      expect(defaultTool).to.equal("place-koi-pond");
+
+      IModelHost.authorizationClient = new AzuriteTest.AuthorizationClient();
+      AzuriteTest.userToken = AzuriteTest.service.userToken.admin;
+
+      const iTwinId = Guid.createValue();
+
+      // __PUBLISH_SECTION_START__ WorkspaceExamples.CreateWorkspaceDb
+      const editor = WorkspaceEditor.construct();
+      const container: EditableWorkspaceContainer = await editor.createNewCloudContainer({
+        // A description of the new WorkspaceContainer.
+        metadata: {
+          label: "trees",
+          description: "trees organized by genus",
+        },
+        // Ownership, access control, and datacenter location are defined by the iTwin.
+        scope: { iTwinId },
+        // The name of the default WorkspaceDb to be created inside the new container.
+        dbName: "cornus",
+        // The manifest to be embedded inside the default WorkspaceDb.
+        manifest: {
+          // A user-facing name for the WorkspaceDb.
+          workspaceName: "Trees: cornus",
+          // A description of the WorkspaceDb's contents and purpose.
+          description: "Trees belonging to the genus cornus",
+          // The name of someone (typically an administrator) who can provide help and information
+          // about this WorkspaceDb.
+          contactName: "Sylvia Wood",
+        },
+      });
+      // __PUBLISH_SECTION_END__
+      
+      expect(container.cloudProps).not.to.be.undefined;
+
+      // __PUBLISH_SECTION_START__ WorkspaceExamples.AddTrees
+      interface TreeResource {
+        commonName: string;
+        hardiness: HardinessRange;
+        light: "full" | "shade" | "partial";
+      }
+
+      function addTree(treeDb: EditableWorkspaceDb, species: string, tree: TreeResource): void {
+        treeDb.addString(species, JSON.stringify(tree));
+      }
+
+      const cornusDb = container.getEditableDb({ dbName: "cornus" });
+
+      addTree(cornusDb, "alternifolia", {
```

Comment:
@kabentley what exactly can you do with the default workspace in a newly-created container? I get `Error: cornus:1.0.0 has been published and is not editable. Make a new version first.`. I understand why that happens, but I don't understand why we create an empty default workspace that can never have content added to it unless we create a new version.

I suppose if we made pre-release dbs editable that would solve the problem. Am I overlooking something here?

Reply:
See my next commit, where I switch to creating a new db and trying to populate it, receiving the same error. It seems the only way to edit a db is to make a new version of an existing one.

Reply:
Switched to creating and populating a new version of the empty db for now.

Reply:
#6811